exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1538 @@
+	formattedPair, err := by.FormatExchangeCurrency(pair, a)
+	if err != nil {
+		return klineItem, err
+	}
+
+	switch a {
+	case asset.Spot:
+		candles, err := by.GetKlines(ctx, formattedPair.String(), by.FormatExchangeKlineInterval(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start, end)
+		if err != nil {
+			return klineItem, err
+		}
+
+		for x := range candles {
+			klineItem.Candles = append(klineItem.Candles, kline.Candle{
+				Time:   candles[x].StartTime,
+				Open:   candles[x].Open,
+				High:   candles[x].High,
+				Low:    candles[x].Low,
+				Close:  candles[x].Close,
+				Volume: candles[x].Volume,
+			})
+		}
+	case asset.CoinMarginedFutures, asset.Futures:
+		candles, err := by.GetFuturesKlineData(ctx, formattedPair, by.FormatExchangeKlineIntervalFutures(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start)
+		if err != nil {
+			return klineItem, err
+		}
+
+		for x := range candles {
+			klineItem.Candles = append(klineItem.Candles, kline.Candle{
+				Time:   time.Unix(candles[x].OpenTime, 0),
+				Open:   candles[x].Open,
+				High:   candles[x].High,
+				Low:    candles[x].Low,
+				Close:  candles[x].Close,
+				Volume: candles[x].Volume,
+			})
+		}
+	case asset.USDTMarginedFutures:
+		candles, err := by.GetUSDTFuturesKlineData(ctx, formattedPair, by.FormatExchangeKlineIntervalFutures(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), start)
+		if err != nil {
+			return klineItem, err
+		}
+
+		for x := range candles {
+			klineItem.Candles = append(klineItem.Candles, kline.Candle{
+				Time:   time.Unix(candles[x].OpenTime, 0),
+				Open:   candles[x].Open,
+				High:   candles[x].High,
+				Low:    candles[x].Low,
+				Close:  candles[x].Close,
+				Volume: candles[x].Volume,
+			})
+		}
+
+	default:
+		return klineItem, fmt.Errorf("assetType not supported: %v", a)
+	}
+
+	klineItem.RemoveOutsideRange(start, end)
+	klineItem.SortCandlesByTimestamp(false)
+	return klineItem, nil
+}
+
+// GetHistoricCandlesExtended returns candles between a time period for a set time interval
+func (by *Bybit) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
+	if err := by.ValidateKline(pair, a, interval); err != nil {
+		return kline.Item{}, err
+	}
+
+	klineItem := kline.Item{
+		Exchange: by.Name,
+		Pair:     pair,
+		Asset:    a,
+		Interval: interval,
+	}
+
+	formattedPair, err := by.FormatExchangeCurrency(pair, a)
+	if err != nil {
+		return klineItem, err
+	}
+
+	dates, err := kline.CalculateCandleDateRanges(start, end, interval, by.Features.Enabled.Kline.ResultLimit)
+	if err != nil {
+		return kline.Item{}, err
+	}
+
+	for x := range dates.Ranges {
+		switch a {
+		case asset.Spot:
+			candles, err := by.GetKlines(ctx, formattedPair.String(), by.FormatExchangeKlineInterval(ctx, interval), int64(by.Features.Enabled.Kline.ResultLimit), dates.Ranges[x].Start.Time, dates.Ranges[x].End.Time)
```

Comment:
This endpoint is still not working. It also has been returning: `[missing data between 2022-05-08 14:02:00 and 2022-05-09 03:43:00]` to highlight this.

GetKlines (docs [here ](https://bybit-exchange.github.io/docs/spot/#t-querykline)) is not matching the documentation:
- You use `start` not `startTime`
- You use `end` not `endTime`

Reply:
I made this change but I still get the same issue. I think the root cause for it is the note which says 
`It only returns the results from the latest 3500 candles regardless of what interval is specified` 
It's present [here](https://bybit-exchange.github.io/docs/spot/#t-querykline)

@gloriousCode

Reply:
Ahh yes I see thank you. I'll have a think about it, so you can move on from this issue for now